service/storage_service.cc

```diff
@@ -615,6 +624,28 @@ topology_node_mutation_builder& topology_node_mutation_builder::set(const char*
     return *this;
 }
 
+template<ConvertibleToSstring S>
+topology_node_mutation_builder& topology_node_mutation_builder::set(const char* cell, const std::set<S>& features) {
+    auto cdef = _builder._s->get_column_definition(cell);
+    assert(cdef);
+    collection_mutation_description cm;
+    if (features.size()) {
+        auto vtype = static_pointer_cast<const set_type_impl>(cdef->type)->get_elements_type();
+
+        cm.cells.reserve(features.size());
+
+        for (auto&& value : features) {
+            cm.cells.emplace_back(vtype->decompose(sstring(value)), atomic_cell::make_live(*bytes_type, _builder._ts, bytes_view()));
+        }
+
+        cm.tomb = tombstone(_builder._ts - 1, gc_clock::now());
+        _r.cells().apply(*cdef, cm.serialize(*cdef->type));
+    } else {
+        del(cell);
```

Comment:
Why do you want to delete by writing an empty set?

Reply:
I based this function on the overload of `topology_node_mutation_builder::set` for the token set which does the same thing. Maybe it would make more sense to write an empty set here... but then I think the overload for tokens should be changed as well.

All in all, I don't think this function will be ever called with an empty feature set, so it doesn't matter what it does in this case. Perhaps I can try to unify this overload with the one for setting tokens.

Reply:
It's cleaner to do the conversion at the caller, rather than duplicate this large function for every type.

Also the original set might be sorted incorrectly. So perhaps the template should create a new set<sstring>, and call a non-template function to create the mutation.

Also: I want one day to be able to write `auto m = cql_statement_as_mutation("UPDATE tab SET my_set = ?", a_set);`

Reply:
This should be fixed now. The setters for both set of features and set of tokens use an internal helper which encapsulates most of the nitty-gritty details and ensures proper order of the values.